sound/soc/sof/ipc4-telemetry.h

```diff
@@ -0,0 +1,68 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __SOUND_SOC_SOF_IPC4_TELEMETRY_H
+#define __SOUND_SOC_SOF_IPC4_TELEMETRY_H
+
+/* Target code */
+enum coredump_tgt_code {
+	COREDUMP_TGT_UNKNOWN = 0,
+	COREDUMP_TGT_X86,
+	COREDUMP_TGT_X86_64,
+	COREDUMP_TGT_ARM_CORTEX_M,
+	COREDUMP_TGT_RISC_V,
+	COREDUMP_TGT_XTENSA,
+};
+
+#define COREDUMP_ARCH_HDR_ID 'A'
+#define COREDUMP_HDR_ID0 'Z'
+#define COREDUMP_HDR_ID1 'E'
+
+/* Coredump header */
+struct coredump_hdr {
+	/* 'Z', 'E' */
+	char id[2];
+
+	/* Header version */
+	u16 hdr_version;
+
+	/* Target code */
+	u16 tgt_code;
+
+	/* Pointer size in Log2 */
+	u8 ptr_size_bits;
+
+	u8 flag;
+
+	/* Coredump Reason given */
+	u32 reason;
+} __packed;
+
+/* Architecture-specific block header */
+struct coredump_arch_hdr {
```

Comment:
can we add prefixes for all those coredump definitions?

Reply:
sure, thanks